### ISSUE TITLE
Archive patient modal does not return focus to when exiting modal

### ISSUE THREAD

**Issue by BobanL** [CDCgov/prime-simplereport#5817]
## Area or Component

[Patients page - Archive patient modal](https://demo.simplereport.gov/app/patients/1?facility=14fab3c1-070e-4e86-8ec7-4a1fc8e13477)
## Quick Summary

When exiting out of the modal via cancel or escape key, focus should be returned to the archive patient button instead of the Patient search box.

### How to Replicate

Click the three dots button next to patient, click the archive patient, and then either click "No, go back" or escape button.

https://github.com/CDCgov/prime-simplereport/assets/10108172/9afe7fc3-e5e7-422d-a50e-1fb8cceb894d

## Accessibility Violation

[Focus modal not returned](https://docs.deque.com/issue-help/1.0.0/en/focus-modal-not-returned)

## Action Requested

On cancel or escape of the modal, focus should be returned back to
